### PR TITLE
* Error handling

### PR DESCRIPTION
**Pull request by Wagnerp**
* Error handling

image

### CODE REVIEW

**Review by giduac — APPROVED** [review]

File: Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonCustomPaletteBase.cs [hunk 1]
```diff
@@ -2249,35 +2249,44 @@ public void ImportWithUpgrade(Stream stream)
...
         {
-            using var reader = new StreamReader(stream);
-            var end = reader.ReadToEnd();
-            reader.Close();
-
-            using (var currentSupportedPaletteSchemaReader = new StreamReader(PaletteSchemaResources.CurrentSupportedPaletteSchema))
+            try
             {
-                using (var xslTextReader = XmlReader.Create(currentSupportedPaletteSchemaReader))
+                using var reader = new StreamReader(stream);
```
Comment (Smurf-IV) [inline Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonCustomPaletteBase.cs:2254]: What do these three lines do? As end is replaced below without being used before?

**Review by Smurf-IV — CHANGES_REQUESTED** [review]

File: Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonCustomPaletteBase.cs (hunk 1 above)
Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonCustomPaletteBase.cs:2254]: Was from this PR (originally) https://github.com/Krypton-Suite/Standard-Toolkit/pull/1077/files#diff-dbc2716418e699c707d51d10fc7acb7c70ece82d5532003fbadfa034adacc60c

Comment (Smurf-IV) [reply Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonCustomPaletteBase.cs:2254]: Not sure what you mean, it's in the code base I reviewed.

Comment (Wagnerp) [reply Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonCustomPaletteBase.cs:2254]: > Not sure what you mean, it's in the code base I reviewed.

@Smurf-IV Not sure why, but it looks differently on mine?

image

**Pull request merged**